config/property/hazelcast_property.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package property
+
+import (
+	"time"
+)
+
+// Hazelcast is a struct for client properties.
+type Hazelcast struct {
```

Comment:
Isn't the name HazelcastProperty is much more suitable for this struct ?

Reply:
In go since dotimport is not used, it will be imported as `property.Hazelcast` vs ` property.HazelcastPropery` do you think it looks bad ? 

This is from go package names:

```
Avoid stutter. Since client code uses the package name as a prefix when referring to the package contents, the names for those contents need not repeat the package name. The HTTP server provided by the http package is called Server, not HTTPServer. Client code refers to this type as http.Server, so there is no ambiguity.
```

Reply:
When I read this source code with type `Hazelcast`, I automatically expect to see methods like  `Hazelcast.newHazelcastInstance/shutdownAll` etc.. 
This is what we are used to see from Java API. 
I get that golang convention is different also. I personally prefer the `HazelcastProperty` instead. 
Lets get more opinion on this ? @asimarslan wdyt ?